### PR TITLE
Add IsOrdered to helperfunction

### PR DESCRIPTION
**Pull request by nehayward**
This will allow for easier testing when checking validating ascending flags are working.

### CODE REVIEW

File: testing_helpers.go [hunk 1]
```diff
@@ -42,3 +44,18 @@ func NewMessageChannelRTMEvent(channelID string, format string, tokens ...interf
 		},
 	}
 }
+
+// IsOrdered asserts that the specified objects are in specified order
+func IsOrdered(t *testing.T, input string, expected ...string) {
```
Comment (nehayward) [inline testing_helpers.go]: Maybe `IsSorted` is a better name. I also debated whether this should be an interface.

File: testing_helpers.go [hunk 2]
```diff
@@ -42,3 +44,18 @@ func NewMessageChannelRTMEvent(channelID string, format string, tokens ...interf
...
+	for _, e := range expected {
```
[the remark this answers is not in the document]
Comment (nehayward) [reply testing_helpers.go]: Yeah I had that initial, but it avoids single entries then. Is that okay?

`IsOrdeded(t, "alpha", "alpaha"` 

Won't throw an error.

File: testing_helpers.go [hunk 3]
```diff
@@ -46,16 +46,22 @@ func NewMessageChannelRTMEvent(channelID string, format string, tokens ...interf
 }
 
 // IsOrdered asserts that the specified objects are in specified order
-func IsOrdered(t *testing.T, input string, expected ...string) {
-	for _, e := range expected {
-		if strings.Index(input, e) < 0 {
-			t.Fatal(fmt.Sprintf("\n%s\ndoes not contain %s", input, e))
+func AssertContainsInOrder(t *testing.T, input string, expected ...string) {
+	for i := 0; i < len(expected); i++ {
+		previous := strings.Index(input, expected[i])
```
Comment (zpatrick) [inline testing_helpers.go]: `previous` doesn't accurately describe this variable - `current` and `next` is more accurate

File: testing_helpers.go [hunk 4]
```diff
@@ -46,16 +46,22 @@ func NewMessageChannelRTMEvent(channelID string, format string, tokens ...interf
...
+		if previous < 0 {
+			t.Fatal(fmt.Sprintf("\n%s\ndoes not contain %s", input, previous))
```
Comment (zpatrick) [inline testing_helpers.go]: use `t.Fatalf` here, works the same as `fmt.Sprintf`

File: testing_helpers.go [hunk 5]
```diff
@@ -46,16 +46,22 @@ func NewMessageChannelRTMEvent(channelID string, format string, tokens ...interf
...
 		}
-	}
 
-	for i := 0; i < len(expected)-1; i++ {
-		if strings.Index(input, expected[i]) > strings.Index(input, expected[i+1]) {
-			t.Fatal(fmt.Sprintf("\n%s\nindex out of order: %s comes after %s", input, expected[i], expected[i+1]))
+		if i < len(expected)-1 {
+			next := strings.Index(input, expected[i+1])
+			if next < 0 {
+				t.Fatal(fmt.Sprintf("\n%s\ndoes not contain %s", input, next))
```
Comment (zpatrick) [inline testing_helpers.go]: ditto

File: testing_helpers.go [hunk 6]
```diff
@@ -46,16 +46,22 @@ func NewMessageChannelRTMEvent(channelID string, format string, tokens ...interf
...
+			}
+
+			if previous > next {
+				t.Fatal(fmt.Sprintf("\n%s\nindex out of order: %s comes after %s", input, previous, next))
```
Comment (zpatrick) [inline testing_helpers.go]: ditto

File: testing_helpers.go (hunk 4 above)
Comment (zpatrick) [reply testing_helpers.go]: also, don't add newlines to the start of these outputs - just keep it consistent with everything else

Comment (zpatrick) [reply testing_helpers.go]: also also, this output isn't printing the correct stuff: it should be something like:
```go
t.Fatalf("%#v does not contain the substring %s", input, expected[i])
```

File: testing_helpers.go (hunk 6 above)
Comment (zpatrick) [reply testing_helpers.go]:
```go
t.Fatalf("%s comes after %s in %#v", expected[i], expected[i+1], input)
```

**Pull request closed without merging**